src/H5Eint.c

```diff
@@ -363,6 +355,82 @@ H5E_term_package(void)
     FUNC_LEAVE_NOAPI(n)
 } /* end H5E_term_package() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5E_user_cb_prepare
```

Comment:
Save and restore the default error reporting settings, to prevent an application from changing them in a callback.